pkg/provider/aws/expose.go

```diff
@@ -0,0 +1,123 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/klothoplatform/klotho/pkg/annotation"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/multierr"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+	"github.com/pkg/errors"
+)
+
+func (a *AWS) GenerateExposeResources(gateway *core.Gateway, result *core.ConstructGraph, dag *core.ResourceGraph) error {
+	err := a.CreateRestApi(gateway, result, dag)
+	return err
+}
+
+func (a *AWS) CreateRestApi(gateway *core.Gateway, result *core.ConstructGraph, dag *core.ResourceGraph) error {
+	var errs multierr.Error
+	api := resources.NewRestApi(a.Config.AppName, gateway)
+	dag.AddResource(api)
+	api_references := []core.AnnotationKey{gateway.AnnotationKey}
+	triggers := map[string]string{}
+
+	resourceMap := map[string]*resources.ApiResource{}
+	for _, route := range gateway.Routes {
+		construct := result.GetConstruct(core.AnnotationKey{ID: route.ExecUnitName, Capability: annotation.ExecutionUnitCapability}.ToId())
+		if construct == nil {
+			errs = append(errs, errors.Errorf("Expected execution unit with id, %s, to exist", route.ExecUnitName))
+			continue
+		}
+		execUnit, ok := construct.(*core.ExecutionUnit)
+		if !ok {
+			errs = append(errs, errors.Errorf("Expected construct with id, %s, to be an execution unit", construct.Id()))
+			continue
+		}
+		api_references = append(api_references, execUnit.Provenance())
+		routeTrigger := fmt.Sprintf("%s:%s:%s", route.ExecUnitName, route.Path, route.Verb)
+		triggers[routeTrigger] = routeTrigger
+
+		segments := strings.Split(route.Path, "/")
```

Comment:
A block comment here about this, and the `for` loop below, would help a lot. I find myself forgetting what the `routh.Path` really is, and then when I see you stripping `:`s and `*`s below, etc, I'm not really sure why.

Basically, I'm not quite sure at a high level what the input is and what processing we need to do with it.

Reply:
to be honest, im not entirely sure why all the time either, so i can definitely add comments. Looking through deploylib very much confused me so likely a good pattern to comment about why we have specific logic